src/workerd/io/actor-cache.c++

```diff
@@ -2322,6 +2322,8 @@ kj::Promise<void> ActorCache::flushImpl(uint retryCount) {
   auto countEntry = [&](Entry& entry) {
     // Counts up the number of operations and RPC message sizes we'll need to cover this entry.
 
+    entry.state = FLUSHING;
```

Comment:
Sure, so we want to only traverse the dirty list[s] once to avoid logic bugs. Since we were implicitly relying on the ordering of the dirty list, we had to do both of our traversals within the same synchronous task. Because we were already required to do both traversals in the same task and the only time we traverse once is if the list is empty, we can mark in the first traversal (and remove the second in a later). Does that make sense?